driver-core/src/main/com/mongodb/internal/operation/CommandOperationHelper.java

```diff
@@ -194,21 +194,24 @@ static BsonDocument executeCommand(final ReadBinding binding, final String datab
                                                  final CommandCreator commandCreator, final Decoder<D> decoder,
                                                  final CommandReadTransformer<D, T> transformer, final boolean retryReads,
                                                  final Connection connection) {
-        BsonDocument command = commandCreator.create(source.getServerDescription(), connection.getDescription());
         MongoException exception;
         try {
-            return executeCommand(database, command, decoder, source, connection, binding.getReadPreference(), transformer,
-                    binding.getSessionContext(), binding.getServerApi());
-        } catch (MongoException e) {
-            exception = e;
-
-            if (!shouldAttemptToRetryRead(retryReads, e)) {
-                if (retryReads) {
-                    logUnableToRetry(command.getFirstKey(), e);
+            BsonDocument command = commandCreator.create(source.getServerDescription(), connection.getDescription());
```

Comment:
This can fail, and if it does we still need to release the connection.  So added an outer try block to handle that case.